### DISCUSSION TITLE
Access max_steps from init function

### DISCUSSION THREAD

**Discussion by jacksonsdean** [FLAMEGPU/FLAMEGPU2#1075] [Q&A] [answered]
Hi there

Our init function starts some parallel processes that would like access to the current simulation config. Most importantly, we were hoping to access the configured maximum number of steps from within the init function. Is this possible?

Something like `const unsigned int max_steps = FLAMEGPU->agentModel.getSimulationConfig().steps;` fails because the agentModel is private within the HostAPI.

**Comment by Robadob** [comment, marked as answer]
It's not currently possible, but that's a good idea, and relates to https://github.com/FLAMEGPU/FLAMEGPU2/issues/1066

I'll add a method to expose the simulation configuration via the host API soon (I can probably implement it tomorrow but it's unlikely to be merged until 14th June) .

As a temporary workaround, you can store the max steps in a global variable which should be accessible from the host function's scope.

**Reply by Robadob** [reply]
PR now created, should be available once merged and will be in the next formal release (not currently scheduled but almost certainly before the end of the year).

https://github.com/FLAMEGPU/FLAMEGPU2/pull/1082

**Reply by mondus** [reply]
PR has now been merged into master
